app/javascript/components/Dictionary.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import convertEntities from '../helpers/convertEntities';
+import createPostBody from '../helpers/createPostBody';
+import renderUnsafeXml from '../helpers/renderUnsafeXml';
+import fetchArray from '../helpers/fetchArray';
+import parseXml from '../helpers/parseXml';
+
+export default class Dictionary extends Component {
+
+  state = {
+    current_search: "",
+    error: false,
+    revealed: false,
+    results_defined: [],
+    results_parsed: ""
+  }
+
+  clearDictionary = (ev) => {
+    ev.preventDefault();
+    this.setState({
+      current_search: "",
+      error: false,
+      results_defined: [],
+      results_parsed: ""
+    });
+  }
+
+  currentSearchUpdate = (ev) => {
+    this.setState({ current_search: ev.target.value });
+  }
+
+  fetchDefined = (xml_string) => {
+    let self = this;
+    // parse XML string as dom nodes so we can extract specific nodes (in this case, a node named "lemma")
+    let dom_lemmata = parseXml(xml_string, "lemma");
+    // prepare to fetch all lemmata against /dictionary by creating array of lemmata (plural of lemma), rerendering as just lemmata, and removing all non-unique entries
+    let arr_lemmata = Array.from(dom_lemmata)
+      .map( entry => entry.textContent )
+      .filter( (value, index, self) => self.indexOf(value) === index );
+    // create array of JSON objects out of db calls using array of lemmata from above
+    return Promise.all(fetchArray(arr_lemmata))
+      .then( responses => self.setState({ results_defined: responses }) )
+      .catch( err => self.setState({ error: true }) );
+  }
+
+  fetchParsed = () => {
```

Comment:
Can you think of a better name for this. Its fetching the parsed word? Maybe fetchParsedWord or something